### DISCUSSION TITLE
Voice Settings not working

### DISCUSSION THREAD

**Discussion by FaecTerr** [stoatchat/discussions#799] [Q&A]
Bug: Once audio settings permission granted, you'll get mic/speaker settings (list), but after switching to different menu and switching back resets permission. Audio settings won't save and it will forget you gave permission. No one can hear you, audio doesn't work and you can't turn your mic on if you are on Voice Channel.

**Comment by artemka2281** [comment]
![image](https://github.com/user-attachments/assets/64f59d5c-2cbe-4e8f-929d-81b50252c0bf)

**Reply by INAAAXXX** [reply]
same shit

**Reply by Kubrix** [reply]
> ![image](https://private-user-images.githubusercontent.com/123476605/374731936-64f59d5c-2cbe-4e8f-929d-81b50252c0bf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDYwNDEzODIsIm5iZiI6MTc0NjA0MTA4MiwicGF0aCI6Ii8xMjM0NzY2MDUvMzc0NzMxOTM2LTY0ZjU5ZDVjLTJjYmUtNGU4Zi05MjlkLTgxYjUwMjUyYzBiZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQzMFQxOTI0NDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kNTI1NTY2YmZjYjZkNGE4ZjY0YmIyYjI3NWIxZmU3ZjgxMjFhZWRjNTczOGZmM2EzNDNmNjJlNTI3MGI5ODEzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.lhfgdDR-wJ7h5mnyk5XlTGXpBCt4txgkVzI0iX1x9Oo)

April 2025:
Still "after switching to different menu and switching back resets permission."
And no way to check microphone/speakers like in Discord

**Comment by mrsarthakgupta** [comment]
Thanks for reporting this, that definitely sounds like a bug in how the client is handling device permissions/state rather than just a one-off browser issue.

From your description it seems like:

You grant audio permissions

Input/output devices show up correctly

As soon as you switch away from the audio/settings view and come back, the client “forgets” that permission was granted and resets the device list

Voice channels then fail because the app thinks there’s no valid mic permission/device

A few things that might help narrow it down for the maintainers:

Does this happen in all browsers/clients you’ve tried, or only one (e.g. Chrome, Firefox, desktop app)?

Is it reproducible every time you join a voice channel, or only after a fresh reload/login?

If you check your browser/OS site permissions, does the mic permission stay “Allowed” even after the app forgets it?

If you haven’t already, it would be helpful to open a bug issue in the main repo with:

exact repro steps

browser/OS/client version

screenshots or a short GIF of the behavior

That should make it easier for the devs to track down why the permission state isn’t being persisted when changing menus.

**Comment by thefossczar** [comment]
same issue, voice channels look exactly like other channels, no UI for voice chat is visible, the perms don't remember that I approved / assigned voice settings. Doesnt work on firefox or chrome.
